nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -64,7 +64,28 @@ public static async Task<ImmutableArray<ProjectDiscoveryResult>> DiscoverAsync(s
 
                     // Get the complete set of dependencies including transitive dependencies.
                     var dependencies = indirectDependencies.Concat(directDependencies).ToImmutableArray();
-                    dependencies = dependencies
+
+                    // // Initialize a new list to hold the new dependencies after splitting the ones with semicolons
+                    var processedDependencies = new List<Dependency>();
+
+                    // add all the dependencies that do not contain a semicolon to the new list
+                    processedDependencies.AddRange(dependencies.Where(d => !d.Name.Contains(";")));
```

Comment:
You can likely combine this section with the part below; it's safe to call `.Split()` on a string without a semicolon, you'll just get the single item back.